lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -116,7 +116,9 @@ def bulk(actions)
                     action.map {|line| LogStash::Json.dump(line)}.join("\n") :
                     LogStash::Json.dump(action)
         as_json << "\n"
-        bulk_responses << bulk_send(body_stream) if (body_stream.size + as_json.bytesize) > TARGET_BULK_BYTES
+        if (body_stream.size + as_json.bytesize) > TARGET_BULK_BYTES
+          bulk_responses << bulk_send(body_stream) unless body_stream.size == 0
```

Comment:
Can you talk me through this fix? We only send if the body is nonzero size, but why would we have a zero-sized body in the first place -- how would this occur?